patches/chromium/fix_put_nsvisualeffectview_before_viewscompositorsuperview.patch

```diff
@@ -0,0 +1,34 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Micha=C5=82=20Pichli=C5=84ski?=
+ <michal.pichlinski@openfin.co>
+Date: Tue, 29 Oct 2024 21:16:29 +0100
+Subject: fix: Put NSVisualEffectView before ViewsCompositorSuperview
+
+Otherwise when using `vibrancy` in `BrowserWindow` NSVisualEffectView
```

Comment:
Chromium does not use `NSVisualEffectView` and `vibrancy` in their codebase, so it is not needed for them, however I'll try to upstream this change.

Reply:
These patches can potentially be avoided if we swizzle the function `sortSubviewsUsingFunction` of `BridgedContentView`. We can override the implementation to sort as needed in Electron.

Take a look at how @gerhardberger was able to do this to override mouse hit behavior: https://github.com/gerhardberger/electron-drag-click/blob/5d6878e308d561c711f22027ba76dad48a8b8143/electron_drag_click.mm#L83-L97

Reply:
Personally I don't like that idea. My fix is dependent on the `SubviewSorter` implementation, it is just adding another special view with higher priority.  If we use overriding implementation of `sortSubviewsUsingFunction` we need to copy `SubviewSorter` implementation and apply to that copy changes which patch introduces. It will be still dependent on the `SubviewSorter` implementation, but without patch we will not notice changes in the original method or we may miss using our version of `SubviewSorter` if Chromium starts using its original implementation somewhere else. It may end up with potentially hidden bug if everything luckily compiles, which could be revealed on the `gclient sync` phase due to conflict during patching. Additionally these overrides look like missing inheritance substituted with inelegant solution. However if you still prefer using `method_setImplementation` and keeping this change fully in the Electron I will replace this patch with `method_setImplementation`.

Reply:
Upstream attempt: https://chromium-review.googlesource.com/c/chromium/src/+/6030552

Reply:
Chromium team is not responding, can we merge this PR with this patch? Or should I use `method_setImplementation` in order to move change outside of the patches (in my opinion this is a bad idea: https://github.com/electron/electron/pull/44628#discussion_r1841084062)

Reply:
ping